lib/plugins/cannulaage.js

```diff
@@ -12,40 +12,87 @@ function init() {
 
-  cage.updateVisualisation = function updateVisualisation (sbx) {
-    var age = 0;
-    var found = false;
-    var treatmentDate = null;
-    var message = '';
+  cage.checkNotifications = function checkNotifications(sbx) {
+  
+    // Just return if no notification times are configured
+  
+  	if (!sbx.extendedSettings.urgent && !sbx.extendedSettings.warn) {
+  	  return;
+  	}
+  
+    var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+	if (!cannulaInfo.checkForAlert) {
+	  return;
+	}
+    
+    // CAGE_WARN=48 CAGE_URGENT=70
+
+    if (cannulaInfo.age === sbx.extendedSettings.urgent || cannulaInfo.age === sbx.extendedSettings.warn )
+    {
+      var title = 'Cannula age ' + cannulaInfo.age +' hours';
+      sbx.notifications.requestNotify({
+          level: 2
+          , title: title
+          , message: title + ', time to change?'
+          , pushoverSound: 'persistent'
+          , plugin: cage
+          , debug: {
+            cannulaInfo: cannulaInfo
+        }
+      });
+      
+    }
+  }
+
+  function frac(f) {
+    return f % 1;
+  }
 
-    _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
+  cage.findLatestTimeChange = function findLatestTimeChange(sbx) {
+  
+  var returnValue = {'message':'', 'found': false, 'age': 0, 'treatmentDate': null, 'checkForAlert': false};
+  
+      _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
       if (treatment.eventType === 'Site Change') {
-        treatmentDate = new Date(treatment.created_at);
-        var hours = Math.round(Math.abs(sbx.time - treatmentDate) / 36e5);
+        returnValue.treatmentDate = new Date(treatment.created_at);
+        var hours = Math.round(Math.abs(sbx.time - returnValue.treatmentDate) / 36e5);
 
-        if (!found) {
-          found = true;
-          age = hours;
+        //allow for around 10 minute period after a full hour during which we'll alert the user
+        returnValue.checkForAlert = frac(Math.abs(sbx.time - returnValue.treatmentDate) / 36e5) < 0.15;
+
+        if (!returnValue.found) {
+          returnValue.found = true;
+          returnValue.age = hours;
         } else {
-          if (hours < age) {
-            age = hours;
+          if (hours < returnValue.age) {
+            returnValue.age = hours;
             if (treatment.notes) {
-              message = treatment.notes;
+              returnValue.message = treatment.notes;
             }
           }
         }
       }
     });
+    
+    return returnValue;
+  }
+  
+  cage.updateVisualisation = function updateVisualisation (sbx) {
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+    var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+    var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+    if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403721/issues/source?bid=2452379&fileBranchId=2457543#l86)